lib/rules/camelcase.js

```diff
@@ -106,8 +106,18 @@ module.exports = {
                         report(node);
                     }
 
-                // Properties have their own rules
-                } else if (node.parent.type === "Property") {
+                // Properties have their own rules, and
+                // AssignmentPattern nodes can be treated like Properties:
+                // e.g.: const { no_camelcased = false } = bar;
+                } else if (node.parent.type === "Property" || node.parent.type === "AssignmentPattern") {
+
+                    if (node.parent.parent.type === "ObjectPattern") {
+                        node.parent.parent.properties.forEach(property => {
```

Comment:
I don't think we need to loop through all the properties here, as this method will already be called on each `Identifier`. This method should worry about a single `Property` node at a time. Additionally, the current implementation will loop through all the destructured properties for each property that exists, which seems unnecessary.